### DISCUSSION TITLE
Integration AKHQ with SchemaRegistry using JsonSchemaProvider not default AvroSchemaProvider

### DISCUSSION THREAD

**Discussion by andrzejpioro** [tchiotludo/akhq#814] [Q&A] [answered]
I'm getting following exception when I try to list schemas in AKHQ:
`2021-09-15 11:54:26,427 ERROR hread-7241 c.k.s.a.AvroSchemaProvider Could not parse Avro schema
org.apache.avro.SchemaParseException: Type not supported: object
	at org.apache.avro.Schema.parse(Schema.java:1734)
	at org.apache.avro.Schema$Parser.parse(Schema.java:1430)
	at org.apache.avro.Schema$Parser.parse(Schema.java:1418)
	at io.confluent.kafka.schemaregistry.avro.AvroSchema.<init>(AvroSchema.java:71)
	at io.confluent.kafka.schemaregistry.avro.AvroSchemaProvider.parseSchema(AvroSchemaProvider.java:54)
	at io.confluent.kafka.schemaregistry.SchemaProvider.parseSchema(SchemaProvider.java:63)
	at org.akhq.repositories.SchemaRegistryRepository.getLatestVersion(SchemaRegistryRepository.java:109)
	at org.akhq.repositories.SchemaRegistryRepository.lambda$toSchemasLatestVersion$1(SchemaRegistryRepository.java:53)
	at java.base/java.util.stream.ReferencePipeline$3$1.accept(Unknown Source)
	at java.base/java.util.ArrayList$SubList$2.forEachRemaining(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.copyInto(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(Unknown Source)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(Unknown Source)
	at java.base/java.util.stream.AbstractPipeline.evaluate(Unknown Source)
	at java.base/java.util.stream.ReferencePipeline.collect(Unknown Source)
	at org.akhq.repositories.SchemaRegistryRepository.toSchemasLatestVersion(SchemaRegistryRepository.java:58)
	at org.akhq.repositories.SchemaRegistryRepository.lambda$list$0(SchemaRegistryRepository.java:42)
	at org.akhq.utils.PagedList.of(PagedList.java:70)
	at org.akhq.repositories.SchemaRegistryRepository.list(SchemaRegistryRepository.java:42)
	at org.akhq.controllers.SchemaController.list(SchemaController.java:51)
	at org.akhq.controllers.$SchemaControllerDefinition$$exec1.invokeInternal(Unknown Source)
	at io.micronaut.context.AbstractExecutableMethod.invoke(AbstractExecutableMethod.java:151)
	at io.micronaut.context.DefaultBeanContext$4.invoke(DefaultBeanContext.java:482)
	at io.micronaut.web.router.AbstractRouteMatch.execute(AbstractRouteMatch.java:303)
	at io.micronaut.web.router.RouteMatch.execute(RouteMatch.java:121)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.emitRouteResponse(RoutingInBoundHandler.java:1533)
	at io.micronaut.http.server.netty.RoutingInBoundHandler.access$2100(RoutingInBoundHandler.java:151)
	at io.micronaut.http.server.netty.RoutingInBoundHandler$6$1.request(RoutingInBoundHandler.java:1500)
	at io.micronaut.http.server.netty.async.ContextCompletionAwareSubscriber.doOnSubscribe(ContextCompletionAwareSubscriber.java:49)
	at io.micronaut.core.async.subscriber.CompletionAwareSubscriber.onSubscribe(CompletionAwareSubscriber.java:38)
	at io.micronaut.configuration.metrics.binder.web.WebMetricsPublisher$1.onSubscribe(WebMetricsPublisher.java:162)
	at org.akhq.middlewares.HttpServerAccessLogFilter$HttpServerPublisher$1.onSubscribe(HttpServerAccessLogFilter.java:133)
	at io.micronaut.http.server.netty.RoutingInBoundHandler$6.doSubscribe(RoutingInBoundHandler.java:1488)
	at io.micronaut.http.server.netty.RoutingInBoundHandler$6.lambda$subscribe$0(RoutingInBoundHandler.java:1482)
	at io.micrometer.core.instrument.composite.CompositeTimer.record(CompositeTimer.java:79)
	at io.micrometer.core.instrument.Timer.lambda$wrap$0(Timer.java:157)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)`

I couldn't find a parameter to specify with types of schema are in schema-registry. Is there a way to use JsonSchemaProvider instead of AvroSchemaProvider? 

Basically all schemas stored in my schema-registry are in JSON format and AKHQ throw same exceptions for each of them.

**Comment by tchiotludo** [comment, marked as answer]
Never used personnally other than avro one, this PR from @bakjos https://github.com/tchiotludo/akhq/pull/760 must handle it. 
Maybe he can explain here the limit of the current implementation ? 
After looking at PR, not really sure for now

**Reply by andrzejpioro** [reply]
@tchiotludo thanks for the link to PR. It explains everything.

**Reply by bakjos** [reply]
Thanks @tchiotludo for pointing this out to me, @andrzejpioro if you use the latest dev image (`tchiotludo/akhq:dev`) if you use containers or the latest master branch from code it should fix your issue, let me know if that's not the case
